### PR TITLE
XR support for BGP AF.

### PR DESCRIPTION
**Pull request by chburket**

### CODE REVIEW

**Comment by glennmatthews** [comment]
Looks like one trivial rubocop failure.

File: lib/puppet/type/cisco_bgp_af.rb [hunk 1]
```diff
@@ -204,8 +204,8 @@ def name
   end
 
   newparam(:afi, namevar: true) do
-    desc 'BGP Address-family AFI (ipv4|ipv6). Valid values are string.'
-    newvalues(:ipv4, :ipv6)
+    desc 'BGP Address-family AFI (ipv4|ipv6|vpnv4|vpnv6). Valid values are string.'
+    newvalues(:ipv4, :ipv6, :vpnv4, :vpnv6)
```
Comment (glennmatthews) [inline lib/puppet/type/cisco_bgp_af.rb]: It doesn't look like your testing does much with vpnv4/vpnv6 at present - are these sufficiently validated to enable at this time?

**Comment by chburket** [comment]
Which file has a failure?  I ran rubocop (and just re-ran) and I don't see anything.

File: lib/puppet/type/cisco_bgp_af.rb (hunk 1 above)
Comment (chburket) [inline lib/puppet/type/cisco_bgp_af.rb]: Good point.  I'll add some tests for these.

**Comment by glennmatthews** [comment]
https://travis-ci.org/cisco/cisco-network-puppet-module/jobs/102188266:

```
tests/beaker_tests/bgpneighbor/test_bgpneighbor_defaults.rb:51:1: W: Unnecessary disabling of Metrics/MethodLength.

# rubocop:disable Style/HashSyntax,Style/ExtraSpacing,Metrics/MethodLength

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by chburket** [comment]
Ah, a file I didn't change was affected by my bump to the method length.  I'll fix.

File: lib/puppet/type/cisco_bgp_af.rb (hunk 1 above)
Comment (chburket) [inline lib/puppet/type/cisco_bgp_af.rb]: @glennmatthews 

I added a test for vpnv4/6 AFIs.

**Comment by glennmatthews** [comment]
You've run the updated test against both NX and XR correct? Otherwise :+1: from me.

**Comment by chburket** [comment]
@glennmatthews 

Yes, beaker tests pass on both.

**Pull request merged**